### PR TITLE
Allow checking the driver versions

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Signed-off-by: Vadim Markovtsev <vadim@sourced.tech>

### CODE REVIEW

File: lookout/core/data_requests.py [hunk 1]
```diff
@@ -16,6 +18,16 @@
 from lookout.core.ports import Type
 
 
+class UnsatisfiedDriverVersionError(Exception):
+    """
+    Error whihc is raised if there is a mismatching Babelfish driver version.
```
Comment (zurk) [inline lookout/core/data_requests.py]:
```suggestion
    An Exception which is raised if there is a mismatching Babelfish driver version.
```

File: lookout/core/data_requests.py [hunk 2]
```diff
@@ -16,6 +18,16 @@
...
+    """
+
+    def __init__(self, mismatched: Iterable[Tuple[str, str]]):
+        """Initialize a new instance of UnsatisfiedDriverVersionError."""
```
Comment (zurk) [inline lookout/core/data_requests.py]:
```suggestion
        """Initialize a new instance of `UnsatisfiedDriverVersionError`."""
```

File: lookout/core/data_requests.py [hunk 3]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
                 bblfsh.aliases.ProtocolServiceStub(self._get_channel())
         return stub
 
+    def check_bblfsh_driver_versions(self, versions: Iterable[str]):
```
Comment (zurk) [inline lookout/core/data_requests.py]:
```suggestion
    def check_bblfsh_driver_versions(self, versions: Iterable[str]) -> None:
```

File: lookout/core/data_requests.py [hunk 4]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
...
+        """
+        Ensure that the Babelfish drivers match the required versions.
+
+        The check is performed by `packaging.version`.
+
+        :param versions: setup.py-like version specifiers, e.g. "javascript==1.3.0".
+        :return: Nothing
```
Comment (zurk) [inline lookout/core/data_requests.py]: I think we can delete these line 
```suggestion
```

File: lookout/core/data_requests.py [hunk 5]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
...
+        :raise UnsatisfiedDriverVersionError: if there is one or more mismatches.
+        """
+        existing = self.get_bblfsh().SupportedLanguages(
+            bblfsh.aliases.SupportedLanguagesRequest()).languages
+        mismatched = []
+        for reqstr in versions:
```
Comment (zurk) [inline lookout/core/data_requests.py]: I think we do not use types in names
How about `requirement`?
```suggestion
        for requirement in versions:
```

File: lookout/core/data_requests.py [hunk 6]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
...
+        Ensure that the Babelfish drivers match the required versions.
+
+        The check is performed by `packaging.version`.
+
+        :param versions: setup.py-like version specifiers, e.g. "javascript==1.3.0".
+        :return: Nothing
+        :raise UnsatisfiedDriverVersionError: if there is one or more mismatches.
+        """
+        existing = self.get_bblfsh().SupportedLanguages(
+            bblfsh.aliases.SupportedLanguagesRequest()).languages
```
Comment (zurk) [inline lookout/core/data_requests.py]: 1 / 3
```suggestion
            bblfsh.aliases.SupportedLanguagesRequest()).languages
        existing = {driver.language: Version(driver.version) for driver in existing}
```

File: lookout/core/data_requests.py [hunk 7]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
...
+        mismatched = []
+        for reqstr in versions:
+            req = Requirement(reqstr)
+            try:
+                ver = Version([d.version for d in existing if d.language == req.name][0])
```
Comment (zurk) [inline lookout/core/data_requests.py]: 2 / 3
```suggestion
                ver = existing[req.name]
```

File: lookout/core/data_requests.py [hunk 8]
```diff
@@ -56,6 +68,31 @@ def get_bblfsh(self) -> bblfsh.aliases.ProtocolServiceStub:
...
+            except IndexError:
```
Comment (zurk) [inline lookout/core/data_requests.py]:
```suggestion
            except KeyError:
```

File: lookout/core/data_requests.py (hunk 4 above)
Comment (vmarkovtsev) [reply lookout/core/data_requests.py]: This is fine

File: lookout/core/data_requests.py (hunk 5 above)
Comment (vmarkovtsev) [reply lookout/core/data_requests.py]: This is fine

**Pull request merged**